Indicators/Vega.cs

```diff
@@ -0,0 +1,221 @@
+        /// <param name="mirrorOption">The mirror option for parity calculation</param>
+        /// <param name="optionModel">The option pricing model used to estimate Vega</param>
+        /// <param name="ivModel">The option pricing model used to estimate IV</param>
+        public Vega(Symbol option, PyObject riskFreeRateModel, decimal dividendYield = 0.0m, Symbol mirrorOption = null,
+            OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes, OptionPricingModelType? ivModel = null)
+            : this($"Vega({option},{mirrorOption},{optionModel})", option, riskFreeRateModel, dividendYield, mirrorOption, optionModel, ivModel)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the Vega class
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="option">The option to be tracked</param>am>
+        /// <param name="riskFreeRate">Risk-free rate, as a constant</param>
+        /// <param name="dividendYield">Dividend yield, as a constant</param>
+        /// <param name="mirrorOption">The mirror option for parity calculation</param>
+        /// <param name="optionModel">The option pricing model used to estimate Vega</param>
+        /// <param name="ivModel">The option pricing model used to estimate IV</param>
+        public Vega(string name, Symbol option, decimal riskFreeRate = 0.05m, decimal dividendYield = 0.0m, Symbol mirrorOption = null, 
+            OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes, OptionPricingModelType? ivModel = null)
+            : base(name, option, riskFreeRate, dividendYield, mirrorOption, optionModel, ivModel)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the Vega class
+        /// </summary>
+        /// <param name="option">The option to be tracked</param>
+        /// <param name="riskFreeRate">Risk-free rate, as a constant</param>
+        /// <param name="dividendYield">Dividend yield, as a constant</param>
+        /// <param name="mirrorOption">The mirror option for parity calculation</param>
+        /// <param name="optionModel">The option pricing model used to estimate Vega</param>
+        /// <param name="ivModel">The option pricing model used to estimate IV</param>
+        public Vega(Symbol option, decimal riskFreeRate = 0.05m, decimal dividendYield = 0.0m, Symbol mirrorOption = null,
+            OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes, OptionPricingModelType? ivModel = null)
+            : this($"Vega({option},{mirrorOption},{optionModel})", option, riskFreeRate, dividendYield, mirrorOption, optionModel, ivModel)
+        {
+        }
+
+        // Calculate the Vega of the option
+        protected override decimal CalculateGreek(DateTime time)
+        {
+            var timeToExpiration = Convert.ToDecimal((Expiry - time).TotalDays) / 365m;
```

Comment:
It seems all the greeks use this value, we might just pass it through as an argument calculated at `OptionGreeksIndicatorBase`, to help reduce duplication